src/phoenix/trace/llama_index/callback.py

```diff
@@ -354,11 +353,21 @@ def _message_payload_to_str(message: Any) -> Optional[str]:
     return str(message)
 
 
-def _get_response_content(response: Any) -> str:
+def _get_response_output(response: Any) -> Iterator[Tuple[str, Any]]:
     """
-    Gets content from response objects. This is needed since the string representation of some
-    response objects includes extra information in addition to the content itself.
+    Gets output from response objects. This is needed since the string representation of some
+    response objects includes extra information in addition to the content itself. In the
+    case of an agent's ChatResponse the output may be a `function_call` object specifying
+    the name of the function to call and the arguments to call it with.
     """
     if isinstance(response, ChatResponse):
-        return response.message.content or ""
-    return str(response)
+        message = response.message
+        if content := message.content:
+            yield OUTPUT_VALUE, content
+            yield OUTPUT_MIME_TYPE, MimeType.TEXT
```

Comment:
I think we might want to consider this to be `llm.output_messages` - is will make the llm IO to be more structured. We could do this in addition to the output so could be a followup.

Reply:
yea, i did notice that, but just wanted to isolate this PR for the time being